### PR TITLE
Properly ignore non-null pipeline creation pointers that should be ignored.

### PR DESCRIPTION
**Pull request by billhollings**
See for reference spec description of `VkGraphicsPipelineCreateInfo`, and CTS test
`dEQP-VK.api.pipeline.pipeline_invalid_pointers_unused_structs.graphics`

Struct pointers that should be ignored under certain conditions, are 
permitted to contain non-null garbage according to the spec and tests.

### CODE REVIEW

File: MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm [hunk 1]
```diff
@@ -1444,9 +1452,9 @@ static MTLVertexFormat mvkAdjustFormatVectorToSize(MTLVertexFormat format, uint3
 												: mvkMTLPrimitiveTopologyClassFromVkPrimitiveTopology(pCreateInfo->pInputAssemblyState->topology);
 	}
 
-    // Color attachments
+    // Color attachments - must ignore bad pColorBlendState pointer if rasterization is disable or subpass has not color attachments
```
Comment (cdavis5e) [inline MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm:1455]:
```suggestion
    // Color attachments - must ignore bad pColorBlendState pointer if rasterization is disabled or subpass has no color attachments
```

File: MoltenVK/MoltenVK/GPUObjects/MVKRenderPass.mm [hunk 2]
```diff
@@ -35,6 +35,17 @@
...
+	for (auto& ca : _colorAttachments) {
+		if (ca.attachment != VK_ATTACHMENT_UNUSED) { return true; }
+	}
+	return false;
+}
+
+bool MVKRenderSubpass::hasDepthStencilAttachment() {
+	return _depthStencilAttachment.attachment != VK_ATTACHMENT_UNUSED;
+}
+
```
Comment (cdavis5e) [inline MoltenVK/MoltenVK/GPUObjects/MVKRenderPass.mm:48]: This is simple enough to be inlined.

**Review by cdavis5e — CHANGES_REQUESTED** [review]
Looks good. Just a few minor nits.

File: MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm (hunk 1 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm:1455]: Fixed.

File: MoltenVK/MoltenVK/GPUObjects/MVKRenderPass.mm (hunk 2 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/GPUObjects/MVKRenderPass.mm:48]: Fixed.

**Review by cdavis5e — APPROVED** [review]

**Pull request merged**